dask/dataframe/io.py

```diff
@@ -452,3 +452,55 @@ def dataframe_from_ctable(x, slc, columns=None, categories=None):
                         np.searchsorted(categories[columns], chunk),
                         categories[columns], True)
         return pd.Series(chunk, name=columns)
+
+
+def from_dask_array(x, columns=None):
+    """ Convert dask Array to dask DataFrame
+
+    Converts a 2d array into a DataFrame and a 1d array into a Series.
+
+    Parameters
+    ----------
+    x: da.Array
+    columns: list or string
+        list of column names if DataFrame, single string if Series
+
+    Example
+    -------
+
+    >>> import dask.array as da
+    >>> import dask.dataframe as dd
+    >>> x = da.ones((4, 2), chunks=(2, 2))
+    >>> df = dd.io.from_dask_array(x, columns=['a', 'b'])
+    >>> df.compute()
+       a  b
+    0  1  1
+    1  1  1
+    2  1  1
+    3  1  1
+    """
+    name = 'from-dask-array' + next(tokens)
+    divisions = [0]
+    for c in x.chunks[0]:
+        divisions.append(divisions[-1] + c)
+
+    index = [(range, a, b) for a, b in zip(divisions[:-1], divisions[1:])]
+
+    if x.ndim == 1:
+        dsk = dict(((name, i), (pd.Series, chunk, ind, x.dtype, columns))
+                for i, (chunk, ind) in enumerate(zip(x._keys(), index)))
+        return Series(merge(x.dask, dsk), name, columns, divisions)
+
+    elif x.ndim == 2:
+        if columns is None:
+            raise ValueError("Must provide columns for DataFrame")
+        if len(columns) != x.shape[1]:
+            raise ValueError("Columns must be the same length as array width")
```

Comment:
I think it would be good to state the observed values for `len(columns)` and `x.shape[1]` in the exception message to help the user quickly pinpoint the source of the problem without having to start a debugger to introspect those values.